### DISCUSSION TITLE
How to add a script block to head in Nuxt 3?

### DISCUSSION THREAD

**Discussion by munjalpatel** [nuxt/nuxt#16109] [Questions] [answered]
Hello,

I simply want to add a `script` block in the `head` tag.

**Example**
```javascript
<script>
    alert('hello, world!');
</script>
```

I spent hours to figure out a solution for something as simple as this.
There are tons of answers about adding `inline` scripts, but none for the script `block` for `Nuxt 3`

How can we do this in `Nuxt 3`?

**Comment by Flanker72** [comment, marked as answer]
Please, have a look at https://v3.nuxtjs.org/guide/features/head-management#meta-components
There is an example with `Style` meta-tag providing `children` property with style block content.
You can use `Script` meta-tag the same way:
```
<template>
  <Script children='console.log("test");' />
</template>
```

Alternatively, you can use `useHead` to provide such a script:
```
<script setup lang="ts">
useHead({
  script: [{ children: 'console.log("test2");' }],
});
</script>
```

And also you may use `nuxt.config` to provide `app.head.script` with `children` attribute on the app level:
```
import { defineNuxtConfig } from "nuxt";
export default defineNuxtConfig({
  app: {
    head: {
      script: [{ children: 'console.log("test3");' }],
    },
  },
});
```

**Reply by Firevvork** [reply]
Who knows that if nobody indicates that directly in the documentation?？

**Comment by slavco86** [comment]
What about asynchronous scripts? Like various analytics widgets?
How do I configure a callback after my script has fetched and loaded all the code from the 3rd party server?

**Reply by ricky11** [reply]
anyone figure this one out?

So far i have : which is able to load the external script, but cant seem to load other tags such as id, api key etc.
head: {
    script: [
      { 
        src: 'https://externalscript.js',
        script: [{children: 'something}]

**Comment by gromann** [comment]
Hi, 
nuxt 3 provides a nice functionality out of the box to do this called useHead. 
With that you can basically add everything you need to add inside the head tags of any page. I have collected and explained some examples of  (also with async scrips) it in this article: [https://www.the-koi.com/projects/everything-you-need-to-know-about-nuxt3-usehead/](https://www.the-koi.com/projects/everything-you-need-to-know-about-nuxt3-usehead/)

Usage: 
```<script setup>
useHead({
  title: "My page with ads",
  // or, instead:
  // titleTemplate: (title) => `My App - ${title}`,
  viewport: "width=device-width, initial-scale=1, maximum-scale=1",
  charset: "utf-8",
  meta: [{ name: "description", content: "My amazing site." }],
  bodyAttrs: {
    class: "test",
  },
  script: [
			{ async: true,
        src: '<https://pagead2.googlesyndication.com/pagead/js/adsbygoogle.js>',
        dataAdClient: "ca-pub-yourId" 
			}, 
			{ src: "anotherScript.js" }
		],
});
</script>```

**Comment by 3ein39** [comment]
What if the script is huge? like the provided newRelic tag for example, converting it to match the syntax to be used in `nuxt.config.js/ts` file is absolutely hard
